spec/controllers/doorkeeper/authorizations_controller_spec.rb

```diff
@@ -269,4 +269,21 @@ def reauthenticate!
       expect(response.status).to be(302)
     end
   end
+
+  describe 'with openid scope but missing response_type and client params' do
+    it 'returns an unsupported_response_type error' do
+      get :new, params: {
+        redirect_uri: application.redirect_uri,
+        scope: 'openid',
+      }
+
+      error_params = {
+        'error' => 'unsupported_response_type',
+        'error_description' => 'The authorization server does not support this response type.'
+      }
+
+      expect(response).to redirect_to build_redirect_uri(error_params)
+      expect(JSON.parse(response.body)).to eq(error_params)
+    end
+  end
```

Comment:
There's a spec that I declined to add because it raises in Doorkeeper proper due to a bad `redirectable?` implementation.  See https://github.com/doorkeeper-gem/doorkeeper/pull/1295

Once that gets resolved, it may be worth re-adding this spec.

```ruby
describe 'with openid scope but missing all necessary params' do
  xit 'returns an unsupported_response_type error' do
    get :new, params: { scope: 'openid' }

    error_params = {
      'error' => 'unsupported_response_type',
      'error_description' => 'The authorization server does not support this response type.'
    }

    expect(response).to redirect_to build_redirect_uri(error_params)
    expect(JSON.parse(response.body)).to eq(error_params)
  end
end
```